tests/common.py

```diff
@@ -858,13 +858,25 @@ class MockPlatform:
 
     def __init__(
         self,
-        setup_platform=None,
-        dependencies=None,
-        platform_schema=None,
-        async_setup_platform=None,
-        async_setup_entry=None,
-        scan_interval=None,
-    ):
+        *,
+        setup_platform: Callable[
```

Comment:
Disable positional arguments, which enabled us to spot incorrect calls with the first parameter being passed as HomeAssistant or as a string.
Onlly affects `setup_platform` as all other arguments are already passed as keyword arguments.